src/utils.js

```diff
@@ -149,4 +149,16 @@ function toReversedProjectionMatrix( projectionMatrix ) {
 
 }
 
-export { arrayMin, arrayMax, arrayNeedsUint32, getTypedArray, createElementNS, createCanvasElement, warnOnce, probeAsync, toNormalizedProjectionMatrix, toReversedProjectionMatrix };
+// referred https://github.com/zloirock/core-js/blob/78474627edab1ab1a1510721185cfc5b66c36770/packages/core-js/internals/global-this.js
+// Three.js nearly always runs on broswers, so it would directly return `self` in most cases.
+// We do not check `window`, because in our situation `self` is enough.
+function getGlobalThis() {
+
+	return typeof self !== 'undefined' ? self : // broswer or Web Worker
+		typeof globalThis !== 'undefined' ? globalThis : // ES2020
+			typeof global !== 'undefined' ? global : // Node.js
+				Function( 'return this' )(); // may break CSP, but it should never run on here
```

Comment:
Do you mind elaborating this comment in more detail?

What is the advantage of using this code instead of `null`? Doesn't `null` better signalize when no global this reference could be returned?